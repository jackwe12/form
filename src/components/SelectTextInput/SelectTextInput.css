.select-text-input-select{
    appearance: none;
    background-image:
    linear-gradient(45deg, transparent 50%, gray 50%),
    linear-gradient(135deg, gray 50%, transparent 50%);
  background-position:
    calc(100% - 6.5px) calc(1em ),
    calc(100% - 4px) calc(1em );
  background-size:
    3px 3px,
    3px 3px;
    background-repeat: no-repeat;

    
}
.select-text-input-select:after{
    position: absolute;
    content: "";
    top: 14px;
    right: 10px;
    width: 0;
    height: 0;
    border: 6px solid transparent;
    border-color: #fff transparent transparent transparent;
}
.change-select-color{
    color:#000!important;
}

.contact select{
    text-align: center;
    color: #d0d0d0;
    border:0;
    border-right: 0.05rem solid #d0d0d0;
    padding-left:8px;
    padding-right:8px;
    border-radius: 0px;
    height: 14.67px;
    background-color: #ffffff;
    height:22px;
    width:3.5rem;
}
select:disabled{
    color:#d0d0d0;
}
select option:first{
    color:#d0d0d0;
}