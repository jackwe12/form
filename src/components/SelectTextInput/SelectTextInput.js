import React from 'react';
import ValidationError from '../ValidationError/ValidationError';
import './SelectTextInput.css';

function SelectTextInput({ name, label, type, value, alert, options, setInputs, placeholder }) {
    return (
        <div className="input-component">
            <label>{label}</label>
            <select  className="select-text-input-select" name={name} type={type} value={value} onChange={setInputs} >
                {options && options.map(option => <option key={option.value} value={option.value} disabled={option.disabled} selected={option.selected}>{option.name}</option>)}
            </select>
            <input id={name} placeholder={placeholder} name={name} type={type} value={value} onChange={setInputs} className={"uk-input" + (alert ? ' uk-form-danger' : '')} />

            <ValidationError content={alert} />
        </div>
    )
}

export default SelectTextInput;

