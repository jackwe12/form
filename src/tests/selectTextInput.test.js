import SelectTextInput from '../components/SelectTextInput/SelectTextInput';
import React from 'react';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });


const mockHandler = jest.fn();
const mockSetSelectColor = jest.fn();
const wrapper = shallow(
    <SelectTextInput 
        setInputs={mockHandler}
        setSelectColor={mockSetSelectColor}
        {...{
        name: "firstName",
        label: "First Name",
        options: [
            { value:'-None', name:'-None', disabled:true, selected:true},
            { value: 'Mr', name: 'Mr' },
            { value: 'Mrs', name: 'Mrs' },
            { value: 'Miss', name: 'Miss' },
            { value: 'Dr', name: 'Dr' }
        ],
        placeholder: "John",
        alert: "",
        }
    }

/>)
const wrapper2 = shallow(
    <SelectTextInput 
        setInputs={mockHandler}
        setSelectColor={mockSetSelectColor}
        {...{
        name: "firstName",
        label: "First Name",
        options: [
            { value:'-None', name:'-None', disabled:true},
            { value: 'Mr', name: 'Mr' },
        ],
        placeholder: "John",
        alert: "error",
        }
    }

/>)

describe('Testing <SelectTextInput/>', ()=>{
    test('snapshot testing',()=>{
        const tree =renderer
            .create(    <SelectTextInput 
                setInputs={mockHandler}
                setSelectColor={mockSetSelectColor}
                {...{
                name: "firstName",
                label: "First Name",
                options: [
                    { value:'-None', name:'-None', disabled:true},
                    { value: 'Mr', name: 'Mr' },
                ],
                placeholder: "John",
                alert: "error",
                }
            }
        />)
            .toJSON();
            expect(tree).toMatchSnapshot();
    })
    
    test('label tag renders correct text',()=>{
        expect(wrapper.find('label').text()).toEqual('First Name');
    });
    test('select tag renders correct prop',()=>{
        expect(wrapper.find('select').prop('defaultValue')).toEqual('-None');
        expect(wrapper.find('select').prop('name')).toEqual('firstName');
    });
    test('option tag renders correct prop, value and disabled in the first one',()=>{
        expect(wrapper.find('option').at(0).text()).toEqual('-None');
        expect(wrapper.find('option').at(0).key()).toEqual('-None');
        expect(wrapper.find('option').at(0).prop('value')).toEqual('-None');
        expect(wrapper.find('option').at(0).prop('disabled')).toEqual(true);


    });
    test('input tag renders correct prop',()=>{
        expect(wrapper.find('input').prop('name')).toEqual('firstName');
        expect(wrapper.find('input').prop('id')).toEqual('firstName');
        expect(wrapper.find('input').prop('placeholder')).toEqual('John');
        expect(wrapper.find('input').prop('value')).toEqual(undefined);

    });
    test('should call mock changeHandler when input changes',()=>{
        const input = wrapper.find('input');
        input.simulate('change');
        expect(mockHandler.mock.calls.length).toBe(1);
    });
    test('should call mock changeHandler when select changes',()=>{
        const select = wrapper.find('select');
        select.simulate('change');
        expect(mockSetSelectColor).toHaveBeenCalled();
    });
    test('when alert is empty, no uk-form-danger class name',()=>{
        expect(wrapper.find('input').hasClass("uk-form-danger")).toBe(false);       
    });
    test('when alert exists, uk-form-danger class name exists',()=>{
        expect(wrapper2.find('input').hasClass("uk-form-danger")).toBe(true);       

    });
});
