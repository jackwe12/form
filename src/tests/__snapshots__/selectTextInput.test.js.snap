// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <SelectTextInput/>  1`] = `
<div
  className="input-component select-text"
>
  <label>
    First Name
  </label>
  <select
    className="select-text-input-select"
    defaultValue="-None"
    name="firstName"
    onChange={[MockFunction]}
  >
    <option
      disabled={true}
      onClick={[MockFunction]}
      value="-None"
    >
      -None
    </option>
    <option
      onClick={[MockFunction]}
      value="Mr"
    >
      Mr
    </option>
  </select>
  <input
    className="uk-input uk-form-danger"
    id="firstName"
    name="firstName"
    onChange={[MockFunction]}
    placeholder="John"
  />
  <div
    className="uk-text-danger"
    style={
      Object {
        "height": "21px",
      }
    }
  >
    <img
      alt="exclamation-circle.png"
      className="exclamation-circle"
      onClick={[Function]}
      src="exclamation-circle.png"
    />
    
  </div>
</div>
`;
